### PR TITLE
(MODULES-7105) Document DSC Resource Distribution

### PR DESCRIPTION
**Pull request by jpogran**
This commit updates the README with documentation on how to install
DSC Resources on target nodes that can be used by the `dsc_lite` module.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,6 +107,43 @@ dsc {'iis_server':
 }
 ~~~
 
+### Distributing DSC Resources
+
+There are several methods to distribute DSC Resources to the target nodes for the `dsc_lite` module to use.
+
+#### PowerShell Gallery
+
+You can choose to install DSC Resources yourself using a mechanism that calls the builtin PowerShell package management system called `PackageManagement`, which uses the `PowerShellGet` module and pulls from the [PowerShell Gallery](https://www.powershellgallery.com). You would be responsible for orchestrating this before Puppet is run on the host, or you could do so using `exec` in your Puppet manifest. This gives you alot of control of the process, but requires you to do alot of the heavy lifting yourself.
```
Comment (RandomNoun7) [inline README.md]: Consider ```control of the process``` to ```control over the process```

Comment (RandomNoun7) [reply README.md]: Consider changing ```do alot of the heavy lifting``` to something a little less metaphorical like ```manage a lot of the complexity of that process```

File: README.md [hunk 2]
```diff
@@ -107,6 +107,43 @@ dsc {'iis_server':
...
+
+~~~puppet
+package { 'xPSDesiredStateConfiguration':
+  ensure   => latest,
+  provider => 'windowspowershell',
+  source   => 'PSGallery',
+}
+~~~
+
+The module supports configuring repository sources and other `PackageManagement` options. For more information please refer to the [forge page](https://forge.puppet.com/hbuckle/powershellmodule).
```
Comment (RandomNoun7) [inline README.md]: Might be nice to call out specifically here that by other sources we mean private and/or on prem packages sources.

File: README.md [hunk 3]
```diff
@@ -107,6 +107,43 @@ dsc {'iis_server':
...
 ~~~
 
+### Distributing DSC Resources
+
+There are several methods to distribute DSC Resources to the target nodes for the `dsc_lite` module to use.
+
+#### PowerShell Gallery
+
+You can choose to install DSC Resources yourself using a mechanism that calls the builtin PowerShell package management system called `PackageManagement`, which uses the `PowerShellGet` module and pulls from the [PowerShell Gallery](https://www.powershellgallery.com). You would be responsible for orchestrating this before Puppet is run on the host, or you could do so using `exec` in your Puppet manifest. This gives you control of the process, but requires you to manage the complexity yourself.
+
```
Comment (glennsarti) [inline README.md]: Given we have examples below, do we need one here?

Comment (jpogran) [reply README.md]: I added an example, but unsure how far we want to go to show all the different things you have to take care of, when there is the hbuckle/powershellmodule already.

File: README.md [hunk 4]
```diff
@@ -107,6 +107,52 @@ dsc {'iis_server':
...
+The following example shows how to install the `xPSDesiredStateConfiguration` DSC Resource, and could be extended to support different DSC Resource names. This example assumes a package repository was configured already.
+
+~~~puppet
+exec { 'xPSDesiredStateConfiguration-Install':
+  command   => 'Install-Module -Name xPSDesiredStateConfiguration -Force',
+  provider => 'powershell',
```
Comment (RandomNoun7) [inline README.md]: NIT: Alignment.

**Pull request merged**